scripts/build-upgrade-image.sh

```diff
@@ -71,6 +71,49 @@ aptly publish repo -skip-contents -skip-signing upgrade-repository
 VERSION=$(dpkg -f "$(find debs/ -name 'delphix-entire-*' | head -n 1)" version)
 sed "s/@@VERSION@@/$VERSION/" <version.info.template >version.info
 
-tar -I pigz -cf "$APPLIANCE_VARIANT.upgrade.tar.gz" version.info -C upgrade-scripts . -C ~/.aptly .
+sha256sum version.info >SHA256SUMS
+for dir in upgrade-scripts ~/.aptly; do
```

Comment:
Hi Prakash, as discussed offline, we could evaluate an alternate approach where we can reduce number of hashes we need to generate as below :

generate a hash of the tarball that we currently use
then sign that hash
bundle the signature, hash, and tarball in another tarball

Advantages would be :
a) small number of hashes to be computed (1 or maybe 2 if we pull out version.info to outer tarball)
b) simpler code both here and in unpack-image

Potential con could be the amount of time it takes to unpack on the engine. Need to compare with the amount of time it takes to generate the large number of hashes (which needs to be repeated during unpack to verify the hash) vs hashing a large file.

Security wise, both approaches achieve what we need.